elastic/kibana
# [Fleet] Support DLM from package-spec

## Summary

Support DLM provided by packages.
When ILM policies are disabled we should install DLM instead that PR does that.

That PR do not address updating/adding DLM to an existing datastream, I will create a follow up PR for that

## Tests

That is cover with unit tests

Manual test, there is no package with DLM yet you can use the elastic_agent package from that branch https://github.com/nchaulet/integrations/tree/test-dlm

* Start a fresh ES and Kibana that use a custom registry using the elastic_agent package with dlm
* Add a fleet server and check the DLM policy is working for elastic_agent.fleet_server logs (data_retention is 2m)

## Todo

- [x] parse from package
- [x] install DLM
- [ ] put dlm for existing datastreams (may come in a different PR)
